tslearn/matrix_profile.py

```diff
@@ -130,21 +148,36 @@ def _transform(self, X, y=None):
         n_ts, sz, d = X.shape
         output_size = sz - self.subsequence_length + 1
         X_transformed = np.empty((n_ts, output_size, 1))
-        scaler = TimeSeriesScalerMeanVariance()
-        band_width = int(np.ceil(self.subsequence_length / 4))
-        for i_ts in range(n_ts):
-            segments = _series_to_segments(X[i_ts], self.subsequence_length)
-            if self.scale:
-                segments = scaler.fit_transform(segments)
-            n_segments = segments.shape[0]
-            segments_2d = segments.reshape((-1, self.subsequence_length * d))
-            dists = squareform(pdist(segments_2d, "euclidean"))
-            band = (np.tri(n_segments, n_segments,
-                           band_width, dtype=np.bool) &
-                    ~np.tri(n_segments, n_segments,
-                            -(band_width + 1), dtype=np.bool))
-            dists[band] = np.inf
-            X_transformed[i_ts] = dists.min(axis=1, keepdims=True)
+
+        if self.algorithm is None:
+            scaler = TimeSeriesScalerMeanVariance()
+            band_width = int(np.ceil(self.subsequence_length / 4))
+            for i_ts in range(n_ts):
+                segments = _series_to_segments(X[i_ts], self.subsequence_length)
+                if self.scale:
+                    segments = scaler.fit_transform(segments)
+                n_segments = segments.shape[0]
+                segments_2d = segments.reshape((-1, self.subsequence_length * d))
+                dists = squareform(pdist(segments_2d, "euclidean"))
+                band = (np.tri(n_segments, n_segments,
+                            band_width, dtype=np.bool) &
+                        ~np.tri(n_segments, n_segments,
+                                -(band_width + 1), dtype=np.bool))
+                dists[band] = np.inf
+                X_transformed[i_ts] = dists.min(axis=1, keepdims=True)
+
+        elif self.algorithm == "stump":
+            import stumpy
+            if d > 1:
+                raise NotImplementedError("Currently only the single"
```

Comment:
This error makes it sound like STUMPY can only support 1D matrix profiles and this is not the case (mstump/mstumped can handle the multi-dimensional case). It's that the multi-dimensional matrix profile has a specific definition (based on the original author's paper) that is somewhat different from the 1D case. 

However, it certainly feels like you are after the 1D uses